src/core/api.rs

```diff
@@ -312,21 +330,23 @@ impl EncryptedHeader {
         usk: &UserSecretKey,
         authentication_data: Option<&[u8]>,
     ) -> Result<Option<CleartextHeader>, Error> {
-        if let Some(symmetric_key) = cover_crypt.decaps(usk, &self.encapsulation)? {
-            let metadata = self
-                .encrypted_metadata
-                .as_ref()
-                .map(|ciphertext| {
-                    cover_crypt.dem_decrypt(&symmetric_key, ciphertext, authentication_data)
-                })
-                .transpose()?;
-            Ok(Some(CleartextHeader {
-                symmetric_key,
-                metadata,
-            }))
-        } else {
-            Ok(None)
-        }
+        CovercryptKEM::decaps(cover_crypt, usk, self.encapsulation.clone()).map(|seed| {
```

Comment:
No clone needed: `CovercryptKem` does not need taking ownership of the encapsulation.
```suggestion
        CovercryptKEM::decaps(cover_crypt, usk, &self.encapsulation).map(|seed| {
```